test/feature-benchmark/scenarios_subscribe.py

```diff
@@ -0,0 +1,134 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from random import getrandbits
+from textwrap import dedent
+
+from materialize.feature_benchmark.measurement_source import MeasurementSource, Td
+from materialize.feature_benchmark.scenario import Scenario
+
+
+class SubscribeParallel(Scenario):
+    """Feature benchmarks related to SUBSCRIBE"""
+
+    SCALE = 2  # So 100 concurrent SUBSCRIBEs by default
+
+    def benchmark(self) -> MeasurementSource:
+        return Td(
+            self.create_subscribe_source()
+            + "\n".join(
+                [
+                    dedent(
+                        f"""
+                        $ postgres-connect name=conn{i} url=postgres://materialize:materialize@${{testdrive.materialize-sql-addr}}
+                        $ postgres-execute connection=conn{i}
+                        # STRICT SERIALIZABLE is affected by #18353
+                        START TRANSACTION ISOLATION LEVEL SERIALIZABLE;
+                        DECLARE c{i} CURSOR FOR SUBSCRIBE s1
+                        """
+                    )
+                    for i in range(0, self.n())
+                ]
+            )
+            + self.insert()
+            # We measure from here ...
+            + dedent(
+                """
+                > SELECT COUNT(*) FROM s1;
+                  /* A */
+                1
+                """
+            )
+            + "\n".join(
+                [
+                    dedent(
+                        f"""
+                        $ postgres-execute connection=conn{i}
+                        FETCH ALL FROM c{i};
+                        """
+                    )
+                    for i in range(0, self.n())
+                ]
+            )
+            # ... to here
+            + dedent(
+                """
+                > SELECT 1
+                  /* B */
+                1
+                """
+            )
+        )
+
+    def create_subscribe_source(self) -> str:
+        assert False, "Scenario needs to provide a source/table definition"
+
+    def create_index(self) -> str:
+        return ""
+
+    def insert(self) -> str:
+        assert False, "Scenario needs to provide insert()"
+
+
+class SubscribeParallelTable(SubscribeParallel):
+    def create_subscribe_source(self) -> str:
+        return dedent(
+            f"""
+             > DROP TABLE IF EXISTS s1;
+             > CREATE TABLE s1 (f1 INTEGER);
```

Comment:
Should we add a different type, for example TEXT, to make it a bit more interesting?

Reply:
yes , actually since the result from the `FETCH` is discarded altogether and does not have to appear in the .td fragment, it can be an arbitrary long or wide result. Will give it a try.